sdp/processors/modify_manifest/data_to_data.py

```diff
@@ -13,15 +13,218 @@
 # limitations under the License.
 
 import collections
+import os
 import re
 from typing import Dict, List
 
+import soundfile as sf
+
 from sdp.logging import logger
 from sdp.processors.base_processor import BaseParallelProcessor, DataEntry
+from sdp.utils.common import ffmpeg_convert
 from sdp.utils.edit_spaces import add_start_end_spaces, remove_extra_spaces
 from sdp.utils.get_diff import get_diff_with_subs_grouped
 
 
+class GetAudioDuration(BaseParallelProcessor):
+    """
+    Processor that computes the duration of the file in audio_filepath_field (using soundfile)
+    and saves the duration in duration_field. If there is an error computing the duration,
+    the duration_field will be updated with the value -1.0.
+
+    Args:
+        audio_file_key (str): Key to get path to wav file.
+        duration_key (str): Key to put to audio duration.
+    Returns:
+        All the same fields as in the input manifest plus duration_field
+    """
+
+    def __init__(
+        self,
+        audio_file_key: str,
+        duration_key: str,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.audio_file_key = audio_file_key
+        self.duration_key = duration_key
+
+    def process_dataset_entry(self, data_entry):
+        audio_filepath = data_entry[self.audio_file_key]
+        try:
+            data, samplerate = sf.read(audio_filepath)
+            data_entry[self.duration_key] = data.shape[0] / samplerate
+        except Exception as e:
+            logger.warning(str(e) + " file: " + audio_filepath)
+            data_entry[self.duration_key] = -1.0
+        return [DataEntry(data=data_entry)]
+
+
+class FfmpegConvert(BaseParallelProcessor):
+    """
+    Processor for converting video or audio files to audio using FFmpeg and updating the dataset with the path to the resampled audio.
+    If id_key is not None it is used as an output file name. If id_key is None the output file name is the same as input file name with different extension
+    and input file name saves to id_key back.
+    Args:
+        resampled_audio_dir (str): The directory to store the resampled audio files.
+        media_file_key (str): The field in the dataset representing the path to the input video or audio files and store the path to the resampled audio files with .wav extension.
+        id_key (str): The field in the dataset representing the unique ID or identifier for each entry. Defaults to None.
+        target_samplerate (int, optional): The target sampling rate for the resampled audio. Defaults to 16000.
+        target_nchannels (int, optional): The target number of channels for the resampled audio. Defaults to 1.
+        **kwargs: Additional keyword arguments to be passed to the base class `BaseParallelProcessor`.
+
+    """
```

Comment:
```suggestion
    Processor for converting video or audio files to audio using FFmpeg and updating the dataset with the path to the resampled audio.
    If ``id_key`` is not None, the output file path will be ``<resampled_audio_dir>/<id_key>.wav``.
    If ``id_key`` is None, the output file path will be ``<resampled_audio_dir>/<input file name without extension>.wav``.

    .. note:: ``id_key`` can be used to create subdirectories inside ``resampled_audio_dir`` (by using forward slashes ``/``). 
        e.g. if ``id_key`` takes the form ``dir_name1/dir_name2/filename``, the output file path will be
        ``<resampled_audio_dir>/dir_name1/dirname2/filename.wav``.

    Args:
        resampled_audio_dir (str): The directory to store the resampled audio files.
        media_file_key (str): The field in the dataset representing the path to the input video or audio files. If ``id_key`` is None, the output file path will be ``<resampled_audio_dir>/<input file name without extension>.wav``.
        id_key (str, optional): The field in the dataset representing the unique ID or identifier for each entry. If ``id_key`` is not None, the output file path will be ``<resampled_audio_dir>/<id_key>.wav``. Defaults to None.
        target_samplerate (int, optional): The target sampling rate for the resampled audio. Defaults to 16000.
        target_nchannels (int, optional): The target number of channels for the resampled audio. Defaults to 1.
        **kwargs: Additional keyword arguments to be passed to the base class `BaseParallelProcessor`.
```

Reply:
Making the docstring clearer. The content inside ".. note::" is dependent on my other code suggestion below